### DISCUSSION TITLE
Mesh adaptivity during one linear step

### DISCUSSION THREAD

**Discussion by am-tc01** [idaholab/moose#22734] [Q&A Meshing] [answered]
Hi,
I am doing steady state thermo-fluid simulations with highly non-linear material properties where the solution changes drastically during one linear iteration. Now since the mesh adaptivity uses variable values from the previous step, it is sometimes difficult to converge. My question is, is it possible to do mesh adaptivity during one linear iteration?

Regards!

**Comment by GiudGiud** [comment, marked as answer]
Hello

no we do not support that currently. We generally would not want to do that because unconverged solution vectors can be completely unphysical, there is no guarantee where the non linear solve lands at every iteration, only the final converged step may be physical. 

Guillaume

**Reply by GiudGiud** [reply]
Is the solve with adaptivity not converging though? You may run multiple cycles of adaptivity per timestep
